### DISCUSSION TITLE
Minion Pillar File Creation

### DISCUSSION THREAD

**Discussion by corycandia** [Security-Onion-Solutions/securityonion#6436] [Unsupported Versions] [answered]
Devs,

Q: Could you be willing to briefly explain your method of automatically creating minion pillar files for configuration?

I am curious how your product is automatically creating minion configuration pillar files.  I'd like to point to your method as 'what right looks like' in some cases.

I have only found the add_minion.sh file, but I can't find what calls it, and only assume it gets passed some variables during execution.  I also see that you guys have soup running scripts against * in the local/minions folder to make modifications to it rather than salt file module functions.

I'm hoping this isn't proprietary.  I've been using SO for a long time and started recently learning SALT with SO being an exceptional example of using it.

**Comment by m0duspwnens** [comment, marked as answer]
Most of the minion pillar files are created when setup is run. If you look at `so-functions` and look for the functions with `_pillar` in the name, you can see that they are written to a temp directory. Then, for remote nodes, they are sent to the manager so that the salt-master can access them.

For the code in soup, that is used to generally modify or add pillar values that are needed for a version upgrade.
